src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java

```diff
@@ -1,170 +1,121 @@
-import uk.gov.ons.ctp.response.sample.domain.repository.SampleAttributesRepository;
+import uk.gov.ons.ctp.response.sample.domain.repository.SampleSummaryRepository;
 import uk.gov.ons.ctp.response.sample.domain.repository.SampleUnitRepository;
-import uk.gov.ons.ctp.response.sample.message.SampleUnitPublisher;
-import uk.gov.ons.ctp.response.sample.representation.SampleSummaryDTO;
-import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO;
-import uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit.SampleAttributes;
+import uk.gov.ons.ctp.response.sample.representation.SampleSummaryDTO.SampleState;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO.SampleUnitState;
+import uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit;
 
-/** Distributes SampleUnits */
+/** Distributes SampleUnits to Collex when requested via job. Retries failures until successful */
 @Component
 @Slf4j
 public class SampleUnitDistributor {
-  public static final String SAMPLEUNIT_DISTRIBUTOR_SPAN = "sampleunitDistributor";
-  public static final String SAMPLEUNIT_DISTRIBUTOR_LIST_ID = "sampleunit";
-  private static final int E = -9999;
-  @Autowired private CollectionExerciseJobRepository collectionExerciseJobRepository;
 
-  @Autowired private SampleUnitRepository sampleUnitRepository;
+  @Autowired private AppConfig appConfig;
 
-  @Autowired private SampleAttributesRepository sampleAttributesRepository;
+  @Autowired private SampleUnitSender sampleUnitSender;
 
-  @Autowired private AppConfig appConfig;
+  @Autowired private CollectionExerciseJobRepository collectionExerciseJobRepository;
 
-  @Autowired private MapperFacade mapperFacade;
+  @Autowired private SampleUnitRepository sampleUnitRepository;
 
-  @Autowired private SampleUnitPublisher sampleUnitPublisher;
+  @Autowired private SampleSummaryRepository sampleSummaryRepository;
 
-  @Autowired
-  @Qualifier("sampleUnitTransitionManager")
-  private StateTransitionManager<SampleUnitDTO.SampleUnitState, SampleUnitDTO.SampleUnitEvent>
-      sampleUnitStateTransitionManager;
+  @Autowired private RedissonClient redissonClient;
 
-  @Autowired private DistributedListManager<Integer> sampleUnitDistributionListManager;
+  @Autowired private SampleUnitMapper sampleUnitMapper;
 
-  /** @return SampleUnitDistributionInfo Information for SampelUnit Distribution */
+  /** Scheduled job for distributing SampleUnits */
+  @Scheduled(fixedDelayString = "#{appConfig.sampleUnitDistribution.delayMilliSeconds}")
   @Transactional
-  public SampleUnitDistributionInfo distribute() {
-    log.info("SampleUnitDistributor is in the house");
-    SampleUnitDistributionInfo distInfo = new SampleUnitDistributionInfo();
+  public void distribute() {
+    List<CollectionExerciseJob> jobs = collectionExerciseJobRepository.findByJobCompleteIsFalse();
+
+    for (CollectionExerciseJob job : jobs) {
+      String uniqueLockName = "SampleCollexJob-" + job.getCollectionExerciseJobPK();
```

Comment:
This needs to be replaced with a StringBuilder or StringBuffer.  The complexity of a that  loop concatenating Strings using + is quadratic.

Reply:
have changed to use a constant, but there's no performance benefit to using a StringBuffer in this instance